common/vskyutil/__init__.py

```diff
@@ -0,0 +1,2105 @@
+    a given altitude, and the interval when it's nighttime.
+
+    Checks are implemented for circumpolar objects that can set and come back up.
+
+    Parameters
+    ----------
+    timeup : astropy Time, can be array
+        Time at which the object rises above an altitude
+    timedown : astropy Time, can be array of the same size as timeup
+        Time at which an object sets below a given altitude
+    evening : astropy Time, can be array of the same size as timeup
+        Time of the beginning of the night
+    morning : astropy Time, can be array of the same size as timeup
+        Time of the ending of the night
+    """
+
+    # all are Times.
+    #  timeup - when the object rises past a given altitle
+    #  timedown - when the object sets past a given altitle
+    #  evening - time of evening twiligth (however defined)
+    #  morning - time of morning twilight
+    # return value will be a TimeDelta
+
+    timeup = Time(np.asarray(timeup.jd), format='jd')
+    scalar_input = False
+    if timeup.ndim == 0:
+        timeup = timeup[None]
+        scalar_input = True
+    timedown = Time(np.asarray(timedown.jd), format='jd')
+    if timedown.ndim == 0:
+        timedown = timedown[None]
+    evening = Time(np.asarray(evening.jd), format='jd')
+    if evening.ndim == 0:
+        evening = evening[None]
+    morning = Time(np.asarray(morning.jd), format='jd')
+    if morning.ndim == 0:
+        morning = morning[None]
+
+    if len(timeup) != len(timedown):
+        print('Error: timeup and timedown must have the same dimensions')
+        return None
+
+    hrsup = TimeDelta(0.0 * np.zeros(len(timeup)), format='jd')
+
+    # up all night
+    ii = np.where(np.logical_and(timeup < evening, timedown > morning))[0][:]
+    if len(ii) != 0:
+        hrsup[ii] = morning[ii] - evening[ii]
+
+    # up and down the same night
+    ii = np.where(np.logical_and(timeup >= evening, timedown <= morning))[0][:]
+    if len(ii) != 0:
+        # rise then set
+        jj = ii[np.where(timedown[ii] >= timeup[ii])[0][:]]
+        if len(jj) != 0:
+            hrsup[jj] = timedown[jj] - timeup[jj]
+        # set then rise the same night
+        kk = ii[np.where(timeup[ii] > timedown[ii])[0][:]]
+        if len(kk) != 0:
+            hrsup[kk] = (timedown[kk] - evening[kk]) + (morning[kk] - timeup[kk])
+
+    # Rise before evening
+    ii = np.where(np.logical_and(timeup < evening, timedown <= morning))[0][:]
+    if len(ii) != 0:
+        hrsup[ii] = timedown[ii] - evening[ii]
+
+    # Set after morning
+    ii = np.where(np.logical_and(timeup >= evening, timedown > morning))[0][:]
+    if len(ii) != 0:
+        hrsup[ii] = morning[ii] - timeup[ii]
+
+    if scalar_input:
+        hrsup = np.squeeze(hrsup)
+    return hrsup
+
+
+def lpsidereal(time, location):
+    """moderate-precision (1 sec) local sidereal time
+
+    Adapted with minimal changes from skycalc routine. Native
```

Comment:
Is this going to be handle by GPP in operations? If not, we might need to revisit this to make it more robust

Reply:
I really have no idea what will happen with this code once GPP is up and running. There is a lot of stuff in `vskyutil` and it would really benefit us, i think, to spend a day cleaning this code up once we verify that everything is working with the original code. I think there are a lot of things we could make much simpler, and the number of warnings this file is producing is huge. I'm wondering if we can clean up a lot here using `np.vectorize`. That's just speculation right now.